src/mono/mono/utils/mono-threads-wasm.c

```diff
@@ -305,6 +309,9 @@ gboolean
 mono_thread_platform_external_eventloop_keepalive_check (void)
 {
 #if defined(HOST_BROWSER) && !defined(DISABLE_THREADS)
+#ifdef ENABLE_CHECKED_BUILD
+	MONO_REQ_GC_SAFE_MODE;
+#endif
```

Comment:
not sure if the `#ifdef` is necessary.  `MONO_REQ_GC_SAFE_MODE;` should be a no-op if we're not in checked mode.

But maybe you need to remove the `;` at the end.  I think some of these older macros aren't really statements